core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionJobPrioritizer.java

```diff
@@ -19,42 +19,127 @@
+  // user tables that have more files that configured system initiated
+  // user tables user initiated
+  // user tables system initiated
+  static final Range<Short> ROOT_TABLE_USER = Range.of((short) 30768, (short) 32767);
+  static final Range<Short> ROOT_TABLE_SYSTEM = Range.of((short) 28768, (short) 30767);
+
+  static final Range<Short> METADATA_TABLE_USER = Range.of((short) 26768, (short) 28767);
+  static final Range<Short> METADATA_TABLE_SYSTEM = Range.of((short) 24768, (short) 26767);
+
+  static final Range<Short> SYSTEM_NS_USER = Range.of((short) 22768, (short) 24767);
+  static final Range<Short> SYSTEM_NS_SYSTEM = Range.of((short) 20768, (short) 22767);
+
+  static final Range<Short> TABLE_OVER_SIZE = Range.of((short) 18768, (short) 20767);
+
+  static final Range<Short> USER_TABLE_USER = Range.of((short) 1, (short) 18767);
+  static final Range<Short> USER_TABLE_SYSTEM = Range.of((short) -32768, (short) 0);
+
+  static {
+    // root table
+    SYSTEM_TABLE_RANGES.put(new Pair<>(RootTable.ID, CompactionKind.USER), ROOT_TABLE_USER);
+    SYSTEM_TABLE_RANGES.put(new Pair<>(RootTable.ID, CompactionKind.SYSTEM), ROOT_TABLE_SYSTEM);
+
+    // metadata table
+    SYSTEM_TABLE_RANGES.put(new Pair<>(MetadataTable.ID, CompactionKind.USER), METADATA_TABLE_USER);
+    SYSTEM_TABLE_RANGES.put(new Pair<>(MetadataTable.ID, CompactionKind.SYSTEM),
+        METADATA_TABLE_SYSTEM);
+
+    // metadata table
+    ACCUMULO_NAMESPACE_RANGES.put(new Pair<>(Namespace.ACCUMULO.id(), CompactionKind.USER),
+        SYSTEM_NS_USER);
+    ACCUMULO_NAMESPACE_RANGES.put(new Pair<>(Namespace.ACCUMULO.id(), CompactionKind.SYSTEM),
+        SYSTEM_NS_SYSTEM);
+  }
+
+  public static short createPriority(final NamespaceId nsId, final TableId tableId,
+      final CompactionKind kind, final int totalFiles, final int compactingFiles,
+      final int maxFilesPerTablet) {
+
+    Objects.requireNonNull(nsId, "nsId cannot be null");
+    Objects.requireNonNull(tableId, "tableId cannot be null");
+    Preconditions.checkArgument(totalFiles >= 0, "totalFiles is negative %s", totalFiles);
+    Preconditions.checkArgument(compactingFiles >= 0, "compactingFiles is negative %s",
+        compactingFiles);
+
+    final Function<Range<Short>,Short> normalPriorityFunction = new Function<>() {
+      @Override
+      public Short apply(Range<Short> f) {
+        return (short) Math.min(f.getMaximum(), f.getMinimum() + totalFiles + compactingFiles);
+      }
+    };
+
+    final Function<Range<Short>,Short> tabletOverSizeFunction = new Function<>() {
+      @Override
+      public Short apply(Range<Short> f) {
+        return (short) Math.min(f.getMaximum(),
+            f.getMinimum() + compactingFiles + (totalFiles - maxFilesPerTablet));
+      }
+    };
+
+    Range<Short> range = null;
+    Function<Range<Short>,Short> func = normalPriorityFunction;
+    if (Namespace.ACCUMULO.id() == nsId) {
+      // Handle system tables
```

Comment:
There could be chop compactions when merging the metadata table and these may end w/ a null range causing an exception. Need to add handling for kind of chop.   Will have to drop this in the 3.1 code.

Reply:
In 74d66b2 I added code to treat CHOP as USER compaction kind.